src/renderer/components/recording-page/recording-visualization.tsx

```diff
@@ -2,134 +2,162 @@ import log from 'electron-log';
+}) => {
   const waveSurferRef = useRef<WaveSurfer>();
   const waveformContainerRef = useRef<HTMLDivElement>(null);
-  const waveformImageRef = useRef<HTMLImageElement>(null);
   const spectrogramContainerRef = useRef<HTMLDivElement>(null);
-  const spectrogramImageRef = useRef<HTMLImageElement>(null);
+  const [, , audioInputDeviceId, , ,] = useAudioInputOutputDevices();
+  const fileMonitor = new FileMonitor(basePath);
+
+  /**
+   * Check to see if the spectrogram plugin is active on the wavesurfer instance.
+   * If the plugin is not active, add and create a new instance of the Spectrogram plugin.
+   */
+  function checkSpectrogramPlugin() {
+    const waveSurfer = waveSurferRef.current;
+
+    if (!waveSurfer) {
+      return;
+    }
+
+    if (!('spectrogram' in waveSurfer.getActivePlugins())) {
+      waveSurfer
+        .addPlugin(
+          SpectrogramPlugin.create({
+            container: spectrogramContainerRef.current!,
+            labels: false,
+          }),
+        )
+        .initPlugin('spectrogram');
+    }
+  }
+
+  /**
+   * Load a .wav file into the waveform instance.
+   */
+  function loadWavFile(filePathToRead: string) {
+    const waveSurfer = waveSurferRef.current;
+
+    if (!waveSurfer) {
+      return noOp();
+    }
+    let cancelled = false;
+
+    (async (): Promise<void> => {
+      if ((await checkFileExistence(filePathToRead)) === 'file') {
+        const data = await readFile(filePathToRead, true);
+        if (!cancelled) {
+          waveSurfer.loadBlob(new Blob([new Uint8Array(data)]));
+          log.info('Triggered waveform update');
+        }
+      }
+    })();
+
+    return (): void => {
+      cancelled = true;
+    };
+  }
 
   useEffect(() => {
     const waveSurfer = WaveSurfer.create({
       interact: false,
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       container: waveformContainerRef.current!,
+      height: 100,
       plugins: [
         SpectrogramPlugin.create({
-          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
           container: spectrogramContainerRef.current!,
-          labels: true,
+          labels: false,
         }),
+        MicrophonePlugin.create({}),
       ],
     });
+
+    fileMonitor.watch(['add', 'unlink', 'change']);
+    fileMonitor.getSubject().subscribe({
+      next: (x: string[]) => {
+        if (x[1].includes(recItemName) && state !== 'recording') {
+          if (x[0] === 'add') {
+            checkSpectrogramPlugin();
+            const filePathToRead = x[1];
+            loadWavFile(filePathToRead);
+          } else if (x[0] === 'unlink') {
+            alert('File ' + x[1] + ' was deleted.');
+            waveSurfer.empty();
+            waveSurfer.destroyPlugin('spectrogram');
+          }
+        }
+      },
+    });
+
     waveSurferRef.current = waveSurfer;
 
-    return (): void => waveSurfer.destroy();
+    return (): void => {
+      waveSurfer.destroy();
+    };
   }, []);
```

Comment:
Do you plan to call this only once? Use `useEffectOnce`.

Reply:
Could you either explain more about what this means? Or can we set up a meeting to talk about this? I'm not totally sure when to use a useEffectOnce (and google didn't show anything when I tried to look it up).

Reply:
A `useEffect` with an empty dependency is going to run exactly once: https://stackoverflow.com/a/53121021
Hence, this is a replacement to make it obvious: https://github.com/streamich/react-use/blob/master/docs/useEffectOnce.md